tests/aws-cpp-sdk-mediastore-data-integration-tests/MediaStoreDataTest.cpp

```diff
@@ -109,7 +109,7 @@ namespace
         static Aws::String GetTestContainerName()
         {
             // "-" is not allowed in container name, convert uuid to its hex.
-            static const std::string suffix = HashingUtils::HexEncode(ByteBuffer(Aws::Utils::UUID::RandomUUID())).c_str();
+            static const std::string suffix = HashingUtils::HexEncode(ByteBuffer(Aws::Utils::UUID::RandomUUID().operator ByteBuffer())).c_str();
```

Comment:
why did you have to change this?
it seems to be an API backward incompatible change.

Reply:
This is because of @DmitriyMusatkin 's request changed for [adding a string to array ctor.](https://github.com/aws/aws-sdk-cpp/pull/2952#discussion_r1607114545). UUID is a special class in that it has a 
[operator Aws::String() const](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/include/aws/core/utils/UUID.h#L36)
_AND_
[operator ByteBuffer() const ](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/include/aws/core/utils/UUID.h#L40)

so when we convert a UUID to byte buf the ctor is ambigious now because there is a ctor with both string and byte buf. Is this a breaking API change? _sort of_? if a class has a operator defined for both String and ByteBuf which i think we can rationalize in making this change as we are the only people likely doing that.